packages/discovery-provider/src/api/v1/helpers.py

```diff
@@ -461,6 +461,34 @@ def extend_playlist(playlist):
     return playlist
 
 
+# Filter out hidden tracks if a non-owner is requesting a public playlist
+# See also: queries/query_helpers.py::filter_hidden_tracks
+def filter_hidden_tracks(playlist, current_user_id):
```

Comment:
curious why this has to happen at the API level? is this not something we can do at the query level?

Reply:
yeah that would be cleaner - but i felt weird manipulating data at the query level. imo the query level should just give you back what is in the db.

Reply:
The `playlist_contents` json column historically was the source of truth for playlist tracks: it's what the client actually writes.

It used to be the only place in the DB where this was stored, but now we have the very nice `playlist_tracks` table.
We _could_ use `playlist_tracks` table and join `tracks` to filter out `is_unlisted` tracks (thus pushing filtering down to query level)... but we'd still have to use `playlist_contents` to inform the ordering.

Anyway... it seems like we _could_ make that change, but since the client is all set up to use `playlist_contents` field anyway... to me doing this filter step in python land seems like the most status-quo compatible way to do it even if it feels a bit hacky.